core/domain/state_domain_test.py

```diff
@@ -4372,6 +4372,136 @@ def _mock_logging_function(msg, *args):
             ]
         )
 
+    def test_get_translatable_fields_for_state_content(self):
+        """Get all translatable fields for state content."""
+
+        state = state_domain.State.create_default_state('state_1')
+        state_content_dict = {
+            'content_id': 'content',
+            'html': '<p>state content html</p>'
+        }
+        state.update_content(
+            state_domain.SubtitledHtml.from_dict(state_content_dict))
+
+        translatable_contents = [
+            translatable_content.content
+            for translatable_content in
+            state.get_translatable_fields().values()
+        ]
+        self.assertItemsEqual(
+            translatable_contents, ['<p>state content html</p>']
+        )
+
+    def test_get_translatable_fields_for_answer_groups(self):
+        """Get all the translatable fields for answer group."""
+
+        state = state_domain.State.create_default_state('state_1')
+        state_answer_group = [state_domain.AnswerGroup(
+            state_domain.Outcome(
+                'destination', state_domain.SubtitledHtml(
+                    'feedback_1', '<p>state outcome html</p>'),
+                False, [], None, None), [
+                    state_domain.RuleSpec(
+                        'Equals', {
+                            'x': {
+                                'contentId': 'rule_input_Equals',
+                                'normalizedStrSet': ['Test rule spec.']
+                                }})
+                    ],
+            [],
+            None
+        )]
+        state.update_interaction_id('TextInput')
+        state.update_interaction_answer_groups(state_answer_group)
+
+        translatable_contents = [
+            translatable_content.content
+            for translatable_content in
+            state.get_translatable_fields().values()
+        ]
+        self.assertItemsEqual(
+            translatable_contents, [
+                '<p>state outcome html</p>',
+                ['Test rule spec.']]
+        )
+
+    def test_get_translatable_fields_for_solution(self):
+        """Get all translatable fields for solution."""
+
+        state = state_domain.State.create_default_state('state_1')
```

Comment:
This still looks like an integration test, i.e, we are testing everything using State. Can we just test the Solution class i.e, create a Solution object and asser solution.get_translatable_fields() ?